src/solver/optim-model-filler/ComponentFiller.cpp

```diff
@@ -396,33 +397,43 @@ void ComponentFiller::addConstraints(const LinearProblemApi::FillContext& ctx)
     }
 }
 
+void ComponentFiller::addStaticObjective(
+  const Optimization::TimeDependentLinearExpression& expression) const
+{
+    auto& pb = optimEntityContainer_.Problem();
+    const auto& solverVariables = optimEntityContainer_.getVariables();
+
+    for (const auto& [index, value]: expression[0])
+    {
+        pb.setObjectiveCoefficient(solverVariables[index].get(), value);
+    }
+}
+
 void ComponentFiller::addObjectives(const LinearProblemApi::FillContext& ctx)
 {
     auto* model = component_.getModel();
-    const auto& solverVariables = optimEntityContainer_.getVariables();
     ReadLinearExpressionVisitor visitor(optimEntityContainer_, ctx, component_);
 
     for (const auto& objective: model->Objectives() | locationFilter())
     {
+        const auto root_node = objective.expression().RootNode();
         const auto linearExpression = visitor.visitMergeDuplicates(
           objective.expression().RootNode());
 
-        auto& pb = optimEntityContainer_.Problem();
-        for (const auto& expr: linearExpression)
+        const auto timeIndex = getConstraintTimeIndex(root_node, component_);
+        if (timeIndex == TimeIndex::VARYING_IN_TIME_ONLY
```

Comment:
Instead of : 
```c++
timeIndex == TimeIndex::VARYING_IN_TIME_ONLY
            || timeIndex == TimeIndex::VARYING_IN_TIME_AND_SCENARIO
```
we should have a : 
```c++
bool isVaryingWithTime(const TimeIndex& timeIndex)
{
    return timeIndex == TimeIndex::VARYING_IN_TIME_ONLY
            || timeIndex == TimeIndex::VARYING_IN_TIME_AND_SCENARIO;
}
```
defined in **TimeIndex.h**

Reply:
By the way : **TimeIndex** is a very misleading name for the associated **enum**.
Applied to an expression / node, It actually represents a variation property or a dependency status.
But **TimeIndex** suggests an index in time, that is a particular hour.
We should rename this whenever possible.